src/algebra/order/smul.lean

```diff
@@ -46,6 +46,42 @@ class ordered_smul (R M : Type*)
 (smul_lt_smul_of_pos : ∀ {a b : M}, ∀ {c : R}, a < b → 0 < c → c • a < c • b)
 (lt_of_smul_lt_smul_of_pos : ∀ {a b : M}, ∀ {c : R}, c • a < c • b → 0 < c → a < b)
 
+namespace order_dual
+
+variables {R M : Type*}
+
+instance [has_scalar R M] : has_scalar R (order_dual M) :=
+{ smul := λ k x, order_dual.rec (λ x', (k • x' : M)) x }
```

Comment:
This should have a simp lemma in terms of `to_dual` if possible.

Reply:
What exactly do you have in mind? I didn't write this, I just moved it from the bottom of the file to prove dual lemmas.